### DISCUSSION TITLE
Bamboo Mua SMTP Port 25

### DISCUSSION THREAD

**Discussion by sujiba** [plausible/analytics#4367] [Self-Hosted Support] [answered]
Hi, 

if I try to invite someone, I get the following error:

```
docker-compose logs plausible -f
plausible-app  | Loading plausible..
plausible-app  | Starting dependencies..
plausible-app  | Starting repos..
plausible-app  | create Plausible.Repo database if it doesn't exist
plausible-app  | create Plausible.IngestRepo database if it doesn't exist
plausible-app  | Creation of Db successful!
plausible-app  | Loading plausible..
plausible-app  | Starting dependencies..
plausible-app  | Starting repos..
plausible-app  | Running migrations for Elixir.Plausible.Repo
plausible-app  | Running migrations for Elixir.Plausible.IngestRepo
plausible-app  | Migrations successful!
plausible-app  | 10:03:59.777 request_id=F-L39RyNi3I66QsAAJJh [error] Failed to send e-mail
```

I've tested the mail delivery with the steps described at the end of https://github.com/plausible/analytics/discussions/4149#discussioncomment-9859126 

```
iex(plausible@68437109bdea)3> Plausible.Mailer.deliver_now(email)
{:error,
 %Mua.TransportError{
   reason: {:tls_alert,
    {:bad_certificate,
     ~c"TLS client: In state wait_cert_cr at ssl_handshake.erl:2131 generated CLIENT ALERT: Fatal - Bad Certificate\n"}}
 }}
```

My configuration:
```
iex(plausible@68437109bdea)1> Application.get_env(:plausible, Plausible.Mailer)
[
  adapter: Bamboo.Mua,
  auth: [username: nil, password: nil],
  relay: "mailrelay.domain.tld",
  port: 25
]
```

plausible-conf.env
```
MAILER_ADAPTER=Bamboo.Mua
MAILER_EMAIL= noreply@domain.tld

SMTP_HOST_ADDR=mailrelay.domain.tld
SMTP_HOST_PORT=25
SMTP_HOST_SSL_ENABLED=false
```

Is it possible to disable tls and ssl? 

Thanks in advance.

**Comment by ruslandoga** [comment]
👋 @sujiba 

> Is it possible to disable tls and ssl?

Not if the relay server advertises STARTTLS in extensions.

Would you be able to share the actual SMTP_HOST_ADDR? I would then check what the problem with the certificate is and suggest possible ways to fix it.

**Comment by sujiba** [comment]
Thank you for your quick response 👍.

The mail relay is unfortunately not accessible over the internet. It is only used for internal mail delivery. Is it possible to ignore / add self signed certificates?

**Reply by ruslandoga** [reply]
Since the error is a "Bad Certificate", it likely means you are already using a self-signed certificate which breaks STARTTLS. If the relay is not accessible over the internet, you might be able to remove that certificate.

**Reply by sujiba** [reply, marked as answer]
A short follow-up:

- Disabled TLS for plausible server in sendmail. See also [How to turn off STARTTLS for internal relaying of emails?](https://unix.stackexchange.com/questions/144989/how-to-turn-off-starttls-for-internal-relaying-of-emails):
  - Try_TLS:IP NO
  - srv_features:IP S
- Reverted back to Bamboo.SMTPAdapter

Now it works again. In the meantime I will search for a solution until Bamboo.SMTPAdapter is removed from plausible.

Many thanks again! I'll close this discussion.

**Reply by ruslandoga** [reply]
Glad you were able to fix it. I wonder why revert back to the Bamboo.SMTPAdapter? It would help to know what Bamboo.Mua couldn't do to improve it further :)

**Reply by sujiba** [reply]
After disabling TLS in sendmail I've gotten this error with Bamboo.Mua:
```
Plausible.Mailer.deliver_now(email)
** (ArgumentError) construction of binary failed: segment 2 of type 'binary': expected a binary but got: nil
    (mua 0.2.0) lib/mua.ex:378: Mua.auth/4
    (mua 0.2.0) lib/mua.ex:121: Mua.easy_send_one/6
    (mua 0.2.0) lib/mua.ex:84: Mua.easy_send_any/6
    (bamboo_mua 0.2.0) lib/bamboo_mua.ex:59: Bamboo.Mua.deliver/2
    (bamboo 2.3.0) lib/bamboo/mailer.ex:208: Bamboo.Mailer.deliver_now/4
    iex:2: (file)
```

**Reply by ruslandoga** [reply]
Ah, I see. Thank you for the follow up!

For reference, it's been fixed in https://github.com/plausible/analytics/pull/4298
